### DISCUSSION TITLE
Trouble Pushing to GitHub

### DISCUSSION THREAD

**Discussion by eyergan** [community/community#52121] [New to GitHub]
I decided to post this in "New to GitHub" because I've searched this and the answers always seems a bit above my understanding. I've been having trouble pushing to github from github desktop, it randomly takes forever on "pushing to origin". I've been trying with my most recent push for days and it's still stuck on that. I've tried closing and restarting it and restarting my computer, but that's not helping.

**Comment by Abdullah-moiz** [comment]
Please explain what kind of project your are uploading and you facing this issue and make sure following 

- Ensure that you have a stable internet connection. Slow internet speeds or intermittent connectivity can cause delays in pushing to GitHub. You can check your internet connection by trying to open a web page or using other online services.
- Make sure there are no merge conflicts or other issues with your Git repository. You can use the command line interface to check for any conflicts by running "git status". If you have any conflicts, you can try to resolve them before you try to push again.
- check if you are uploading unnecessary folders like node module that is a packages folder installed while installing different packages in node js, react, etc so make sure you are correctly ignoring it in gitignore file you can create a gitignore file using https://www.toptal.com/developers/gitignore
